### PR TITLE
front/nimconf: add exported version of readConfigFile

### PR DESCRIPTION
**Pull request by alaviss**
## Summary
This API is used by [nimph](https://github.com/disruptek/nimph) to read
configuration files for modification.

## Details
* This is a part of the effort to get nimph running with nimskull

### CODE REVIEW

File: compiler/front/nimconf.nim [hunk 1]
```diff
@@ -502,4 +501,10 @@ proc loadConfigs*(
                              stopOnError: stopOnError)
   parser.loadConfigs(cfg, cache)
 
-  
+proc readConfigFile*(
+  filename: AbsoluteFile, cache: IdentCache,
+  conf: ConfigRef, evtHandler: NimConfEvtWriter
+): bool {.inline.} =
```
Comment (saem) [inline compiler/front/nimconf.nim:507]:
```suggestion
): bool {.inline.} =
  # created and exported for `nimph`
```

**Review by saem — COMMENTED** [review]
Minor suggestion nonblocking

**Comment by disruptek** [comment]
Probably a good stop-gap measure, but ideally, we expose everything we need via json and reduce that list where possible.

**Comment by alaviss** [comment]
> Probably a good stop-gap measure, but ideally, we expose everything we need via json and reduce that list where possible.

How would that work? Like adding a `parsecfg` command or something?

**Pull request merged**